packages/client/hmi-client/src/components/model-template/tera-model-template-editor.vue

```diff
@@ -352,33 +379,35 @@ function mouseUpdate(event: MouseEvent) {
 	prevY = event.y;
 }
 
-function amrToTemplates() {
-	kernelManager.sendMessage('amr_to_templates', {}).on('amr_to_templates_response', (d) => {
-		// FIXME: Model templates are passed no junctions yet
-
-		// Insert template card data into template models
-		let yPos = 100;
-		const templateModelsWithCards = d.content.templates.map((modelTemplate: any) => {
-			modelTemplate.metadata.templateCard = {
-				id: modelTemplate.header.name,
-				name: modelTemplate.header.name,
-				x: 100,
-				y: yPos
-			} as ModelTemplateCard;
-
-			yPos += 200;
-			return modelTemplate;
-		});
+async function initializeBeakerKernel(isModelLoaded = false) {
+	if (kernelManager.jupyterSession) kernelManager.shutdown();
 
-		templateModelsWithCards.forEach((templateModel: any) =>
-			modelTemplatingService.addCard(decomposedTemplates.value, templateModel)
-		);
-	});
+	try {
+		const context = {
+			context: 'mira_model_edit',
+			language: 'python3',
+			context_info: {
+				id: isModelLoaded ? props.model?.id ?? '' : ''
+			}
+		};
+		await kernelManager.init('beaker_kernel', 'Beaker Kernel', context);
+		// Create decomposed view from model
+		if (isModelLoaded && props.model) {
+			modelTemplatingService.flattenedToDecomposed(
+				decomposedTemplates.value,
+				kernelManager,
+				outputCode,
+				interpolatePointsForCurve
+			);
+		}
+	} catch (error) {
+		logger.error(`Error initializing Jupyter session: ${error}`);
+	}
 }
 
 watch(
 	() => props.model,
-	async () => {
+	() => {
 		if (props.model) {
 			// Create flattened view of model
 			const flattenedModel: any = cloneDeep(props.model);
```

Comment:
Sorry not sure where is the best place to add this observation:

To what extend does is the flattened-view editing functional? I can drag templates onto the flattened view, but the templates disappear, however, if I switch to the decomposed view the templates I had created would be there.

Reply:
I somehow wasn't thinking about editing the flattened view, just made a fix so cards are added in there properly. When cards are linked in the flattened view it's supposed to merge the cards once they are connected.

Reply:
^ adding to flattened template should be its own new issue